src/Datadog.Trace/Configuration/ConfigurationSourceExtensions.cs

```diff
@@ -0,0 +1,51 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Datadog.Trace.Configuration
+{
+    internal static class ConfigurationSourceExtensions
+    {
+        private const char Separator = ';';
+
+        public static IEnumerable<string> GetStrings(this IConfigurationSource source, string key)
+        {
+            return source?.GetString(key)
+                ?.Split(new[] { Separator }, StringSplitOptions.RemoveEmptyEntries)
+                ?? Enumerable.Empty<string>();
+        }
+
+        public static TEnum GetTypedValue<TEnum>(this IConfigurationSource source, string key)
```

Comment:
how we would know that an invalid value was provided. [more context](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/102#discussion_r607774012)